### PR TITLE
Anomaly Core Edits

### PR DESCRIPTION
**Pull request by Erikafox**
## About The Pull Request

most anomaly cores now grind down into something (usually on the rare side)
plasmasouls will no longer turn into a raging inferno if shot at
new detonate effect for phantom cores and plasmasoul cores.

## Changelog

:cl:
add: new detonate effect for phantom cores and plasmasoul cores.
add: Anomaly cores now grind down into something.
balance: plasmasouls suck less
/:cl:

### CODE REVIEW

File: code/modules/assembly/anomalies.dm [hunk 1]
```diff
@@ -173,6 +198,7 @@
 /obj/item/assembly/signaler/anomaly/tvstatic
 	name = "\improper static anomaly core"
 	desc = "The stabilized core of a static anomaly. Your head hurts just staring at it"
+	grind_results = list(/datum/reagent/three_eye = 5)
```
Comment (Zevotech) [inline code/modules/assembly/anomalies.dm:201]: erika what the fuck. thats Three Eye.

Comment (Erikafox) [reply code/modules/assembly/anomalies.dm:201]: yeah i thought it'd be cool for the anomaly that gives you brain damage.

Comment (Zevotech) [reply code/modules/assembly/anomalies.dm:201]: its funny but i dont think you should use the Admin Meme Chemical That Unavoidably Kills You And Makes You See Green. idk im not a maintainer.

Comment (firebudgy) [reply code/modules/assembly/anomalies.dm:201]: It literally is permanent and cannot be undone, even by aheals. Not a good idea

Comment (Erikafox) [reply code/modules/assembly/anomalies.dm:201]: a: who the fuck ported this
b: who the fuck let this stay in here this long

Comment (Erikafox) [reply code/modules/assembly/anomalies.dm:201]: yeah okay so i went and read the code and it only does that if you overdose. which is stupid. but something to address in another pr. you'd need to find 5 static anomalies to get enough to overdose yourself like this

Comment (Erikafox) [reply code/modules/assembly/anomalies.dm:201]: also the code is broken its not supposed to turn you into the lime man forever.

Comment (Erikafox) [reply code/modules/assembly/anomalies.dm:201]: okay so - i tested it and it did clear itself. idk what about overdose makes you turn into a citrus fruit, but unless a maintainer says nuh uh I think this is okay.
It's a rarer anomaly and a small amount of what is ostensibly a drug

Comment (MarkSuckerberg) [reply code/modules/assembly/anomalies.dm:201]: the funny :+1:

File: code/modules/assembly/anomalies.dm [hunk 2]
```diff
@@ -231,14 +263,16 @@
 		/obj/effect/gibspawner/generic/animal
 	))
 	visible_message(span_warning("[src] erupts into a fountain of gore and viserca!"))
-	new mess(src.loc)
+	for(var/i in range(0,4))
```
Comment (MarkSuckerberg) [inline code/modules/assembly/anomalies.dm]:
```suggestion
	for(var/i in 1 to 4)
```

File: code/modules/assembly/anomalies.dm [hunk 3]
```diff
@@ -117,25 +123,42 @@
...
-	visible_message(span_warning("[src] rapidly delaminates, leaving behind trace vapors!"))
+	visible_message(span_warning("[src] rapidly delaminates into plasma!"))
+	atmos_spawn_air("plasma=500;TEMP=[T20C]")
 	..()
 
 ///Pulsar Anomaly
 /obj/item/assembly/signaler/anomaly/pulsar
 	name = "\improper pulsar anomaly core"
 	desc = "The stabilized core of a pulsar anomaly. Electromagnetic crackles come off it."
+	grind_results = list(/datum/reagent/iron = 25, /datum/reagent/uranium = 25)
```
Comment (MarkSuckerberg) [inline code/modules/assembly/anomalies.dm:161]: are you Sure you want to do this to people

File: code/modules/assembly/anomalies.dm [hunk 4]
```diff
@@ -90,15 +93,17 @@
 	name = "\improper hallucination anomaly core"
 	desc = "The stabilized core of a hallucination anomaly. It's never entirely there."
 	icon_state = "hallucination_core"
+	grind_results = list(/datum/reagent/drug/space_drugs = 20) //i cannot figure out what to do with this for the life of me
```
Comment (MarkSuckerberg) [inline code/modules/assembly/anomalies.dm]: why not mindbreaker?

**Review by MarkSuckerberg — CHANGES_REQUESTED** [review]

File: code/modules/assembly/anomalies.dm (hunk 4 above)
Comment (MarkSuckerberg) [reply code/modules/assembly/anomalies.dm]: andor mushroom hallucinogen

File: code/modules/assembly/anomalies.dm (hunk 3 above)
Comment (Erikafox) [reply code/modules/assembly/anomalies.dm:161]: I could be convinced otherwise.

File: code/modules/assembly/anomalies.dm (hunk 4 above)
Comment (Erikafox) [reply code/modules/assembly/anomalies.dm]: you are way smarter than i am

File: code/modules/assembly/anomalies.dm (hunk 3 above)
Comment (MarkSuckerberg) [reply code/modules/assembly/anomalies.dm:161]: I do not see it lasting long but it would be funny

**Review by MarkSuckerberg — APPROVED** [review]

**Pull request merged**